src/biotite/database/uniprot/query.py

```diff
@@ -0,0 +1,201 @@
+    combination via 'AND', 'OR' or 'NOT'. A query consists of a search
+    term and an optional field.
+
+    A list of available search fields with description can be found
+    `here <https://www.uniprot.org/help/query-fields>`_.
+
+    Parameters
+    ----------
+    field : str
+        The field to search the term in.
+       The list of possible fields and the required search term
+       formatting can be found
+       `here <https://www.uniprot.org/help/query-fields>`_.
+    term: str
+       The search term.
+    """
+
+    # Field identifiers are taken from
+    # https://www.uniprot.org/help/query-fields
+    _fields = [
+        "accession", "active", "annotation", "author", "cdantigen", "chebi", "citation", "cluster", "count", "created",
+        "database", "ec", "evidence", "existence", "family", "fragment", "gene", "gene_exact", "goa", "host", "id",
+        "inchikey", "inn", "interactor", "keyword", "length", "lineage", "mass", "method", "mnemonic", "modified",
+        "name", "organelle", "organism", "plasmid", "proteome", "proteomecomponent", "replaces", "reviewed", "scope",
+        "sequence", "sequence_modified", "strain", "taxonomy", "tissue", "web"
+    ]
+
+    def __init__(self, field, term):
+        super().__init__()
+        if field not in SimpleQuery._fields:
+            raise ValueError(f"Unknown field identifier '{field}'")
+        if not _check_brackets(term):
+            raise ValueError(
+                f"Query term contains illegal number of round brackets ( ) and/or square brackets [ ]"
+            )
+        for invalid_string in \
+                ['"', "AND", "OR", "NOT", "\t", "\n"]:
+            if invalid_string in term:
+                raise ValueError(
+                    f"Query contains illegal term {invalid_string}"
+                )
+        if " " in term:
+            term = f'"{term}"'
+        self._field = field
+        self._term = term
+
+    def __str__(self):
+        return f"{self._field}:{self._term}"
+
+
+def search(query, number=10):
+    """
+    Get all UniProt IDs that meet the given query requirements,
+    via the UniProt search service.
+
+    This function requires an internet connection.
+
+    Parameters
+    ----------
+    query : Query
+        The search query.
+    number : int
+        The maximum number of IDs that are obtained.
+
+    Returns
+    -------
+    ids : list of str
+        A list of strings containing all UniProt IDs
+        that meet the query requirements.
+
+    Warnings
+    --------
+    Even if you give valid input to this function, in rare cases the
+    database might return no or malformed data to you.
+    In these cases the request should be retried.
+    When the issue occurs repeatedly, the error is probably in your
+    input.
+
+    Notes
+    -----
+    A list of available search fields with description can be found
+    `here <https://www.uniprot.org/help/query-fields>`_.
+    """
```

Comment:
I think an example code snippet would be helpful here, so the user can additionally see how combination of the `SimpleQuery` works.